test/models/user.test.js

```diff
@@ -0,0 +1,180 @@
+require('../../models/User');
+require('../../models/Posts')
+require('sinon-mongoose');
+const mongoose = require('mongoose');
+const chai = require('chai');
+const sinon = require('sinon');
+const UserFactory = require('../factories/users');
+const NotificationFactory = require('../factories/notifications');
+const bcrypt = require('bcryptjs');
+
+const expect = chai.expect;
+const User = mongoose.model('User');
+const Post = mongoose.model('Post')
+const Notification = mongoose.model('Notification')
+const fred = UserFactory.fred
+
+describe("Users", () => {
+	before(() => {
+		const save = sinon.stub(User.prototype, 'save').resolves(this);
+	});
+
+	describe("can be created", () => {
+		it('with no errors', done => {
+			const changePasswordStub = sinon.stub(User.prototype, 'changePassword').resolves(fred);
+			
+			User.create(fred).then(result => {
```

Comment:
Wouldn't we want to be testing from the API endpoint instead of from the schema? It would give us much better coverage if the test case was actually connecting to the server via HTTP requests

Reply:
This is just testing the logic of the model. Its just basic unit testing, testing the smallest possible unit of code.

Testing controllers will be next, and falls more into the category of integration testing. I'm planning on using [supertest](https://github.com/visionmedia/supertest) to do so.